drivers/audio/audio_null.c

```diff
@@ -771,11 +772,12 @@ static int null_ioctl(FAR struct audio_lowerhalf_s *dev, int cmd,
 #endif
 
       default:
+        ret = -EINVAL;
```

Comment:
This  NuttX convention is that if the ioctl functions does not recognize the command, it returns ENOTTY, not EINVAL.

See https://en.wikipedia.org/wiki/Not_a_typewriter which is how NutX does things in ioctls too.

Reply:
> This NuttX convention is that if the ioctl functions does not recognize the command, it returns ENOTTY, not EINVAL.
> 
> See https://en.wikipedia.org/wiki/Not_a_typewriter which is how NutX does things in ioctls too.

Hi @patacongo,

Thanks for the comments. 
Actually, I checked it with 'man ioctl' on Ubuntu18.04 and it says that

```
ERRORS
       EBADF  fd is not a valid file descriptor.
       EFAULT argp references an inaccessible memory area.
       EINVAL request or argp is not valid.
       ENOTTY fd is not associated with a character special device.
       ENOTTY The specified request does not apply to the kind of object that the file descriptor fd references.
```

So I thought we should return EINVAL.
However in NuttX should we return ENOTTY instead of EINVAL?

Reply:
Also https://pubs.opengroup.org/onlinepubs/009695399/functions/ioctl.html

But currently every other NuttX ioctl function returns ENOTTY in that case.  This would be an exception.

Reply:
> Also https://pubs.opengroup.org/onlinepubs/009695399/functions/ioctl.html
> 
> But currently every other NuttX ioctl function returns ENOTTY in that case. This would be an exception.

OK, I will modify the code and follow the NuttX's error code convention.

Reply:
I've just pushed with -f.

Reply:
Here is Linus Torvalds interpretation of that exact same language:  https://lore.kernel.org/patchwork/patch/258361/  I agree with Linus